server/services/licenceService.ts

```diff
@@ -159,12 +160,23 @@ export default class LicenceService {
     const token = await this.hmppsAuthClient.getSystemClientToken(username)
 
     const requestBody = {
-      data: Object.keys(formData).map((key, index) => {
-        return {
-          field: key,
-          value: formData[key],
-          sequence: index,
+      data: Object.keys(formData).flatMap((key, index) => {
+        const build = (value: string, i?: number) => {
+          return {
+            field: key,
+            value,
+            sequence: i || index,
+          }
+        }
+        if (Array.isArray(formData[key])) {
+          return formData[key].map((v: string, i: number) => {
+            return build(v, i)
+          })
+        }
+        if (formData[key] instanceof Stringable) {
+          return build(formData[key].stringify())
         }
+        return build(formData[key])
       }),
```

Comment:
Perhaps worth a few comments around this part? e.g. Why some formData elements are arrays, others Stringable, others neither?

Reply:
Added a comment now and rejigged the logic a bit!